### PR TITLE
feat: implement core AmaltheaSession crd elements creation

### LINKED ISSUE #598: Core objects creation

### PR DESCRIPTION
**Pull request by sgaist**
## Describe your changes

This PR implements the creation part of the core elements of the new AmaltheaSession CRD.

This includes:

- StatefulSet
- Matching Service
- Matching Ingress

## Issue ticket number and link

Close #598 
Part of https://github.com/SwissDataScienceCenter/renku/issues/3679

### CODE REVIEW

**Review by olevski — CHANGES_REQUESTED** [review]
Thanks @sgaist this looks really good. 

But lets do the following refactoring to keep the reconcile loop a bit cleaner.

Create a struct that can hold all of Amaltheas children resources. I.e.

```go
type AmaltheaChildren struct {
	Ingress *networkingv1.Ingress
	Service v1.Service
	StatefulSet appsv1.StatefulSet
	PVC v1.PersistentVolumeClaim
}
```

Then we have a function on the CR to generate this from the manifest:

```go
func (cr *AmaltheaSession) Children() AmaltheaChildren {
    ...
}
```

Then we can use the output of this function to diff the current state against what is desired based on the spec. This function would always return what is desired based on the spec.

File: internal/controller/amaltheasession_controller.go
Comment (sgaist) [reply internal/controller/amaltheasession_controller.go]: Done

File: api/v1alpha1/amaltheasession_types.go
Comment (sgaist) [inline api/v1alpha1/amaltheasession_types.go:306]: @olevski Should we also make these free functions methods of `AmaltheaSession` struct ?

One reason would be to make their call more intuitive when outside of this file when not needing all of the children for example when checking the pod created by the `StatefulSet`

Comment (olevski) [reply api/v1alpha1/amaltheasession_types.go:306]: Yes that would be really nice! After that I will approve and we can merge.

Comment (sgaist) [reply api/v1alpha1/amaltheasession_types.go]: Done !

**Review by olevski — APPROVED** [review]

**Pull request merged**
